### PR TITLE
rework the acceptance update scenario to avoid using snapshort/restore

### PR DESCRIPTION
**Pull request by purbon**
This might be a temporary solution for ssh issues in vagrant and ubuntu hosts, see https://github.com/elastic/logstash/issues/5324#issuecomment-224235456 for more details.

@ph @untergeek as I can not reproduce this in my machine, can you guys please validate this workaround is okey. Thanks a lot.

### CODE REVIEW

File: qa/acceptance/spec/shared_examples/updated.rb [hunk 1]
```diff
@@ -4,19 +4,22 @@
 # This test checks if the current package could used to update from the latest version released.
 RSpec.shared_examples "updated" do |logstash|
 
-  before (:all) { logstash.snapshot }
-  after  (:all) { logstash.restore }
+  before(:all) { logstash.uninstall }
```
Comment (ph) [inline qa/acceptance/spec/shared_examples/updated.rb]: Would that command fails  if the package was already uninstalled?

File: qa/acceptance/spec/shared_examples/updated.rb [hunk 2]
```diff
@@ -4,19 +4,22 @@
...
+  after(:all)  do
+    logstash.stop_service # make sure the service is stopped
+    logstash.uninstall #remove the pacakge to keep uniform state
```
Comment (ph) [inline qa/acceptance/spec/shared_examples/updated.rb]: pacakge -> package.

**Comment by ph** [comment]
I am OK do the uninstall/install dance if we cannot make the snapshot work for now :(

File: qa/acceptance/spec/shared_examples/updated.rb (hunk 1 above)
Comment (purbon) [inline qa/acceptance/spec/shared_examples/updated.rb]: is basically error free for now, it will not rise any error.

**Comment by purbon** [comment]
we have to try and see if this workaround actually "fix" the issue, just looking for a resonable workaround for this problem 👍

**Comment by purbon** [comment]
Tested this in a machine similar to our logstash-ci and faced the same issue with master branch, see

```bash
purbon@slave-342015:~/logstash/qa$ bundle exec rake qa:acceptance:single["ubuntu-1404"]
[Acceptance specs] running on ["ubuntu-1404"]
/home/purbon/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/net-ssh-2.9.4/lib/net/ssh/transport/session.rb:67:in `initialize': Object#timeout is deprecated, use Timeout.timeout instead.
/home/purbon/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/net-ssh-2.9.4/lib/net/ssh/transport/session.rb:84:in `initialize': Object#timeout is deprecated, use Timeout.timeout instead.
.....
An error occurred in an `after(:context)` hook.
  LogStash::CommandExecutor::CommandError: CMD: vagrant snapshot restore ubuntu-1404 ubuntu-1404-snapshot STDERR: Timed out while waiting for the machine to boot. This means that
Vagrant was unable to communicate with the guest machine within
the configured ("config.vm.boot_timeout" value) time period.

If you look above, you should be able to see the error(s) that
Vagrant had when attempting to connect to the machine. These errors
are usually good hints as to what may be wrong.

If you're using a custom box, make sure that networking is properly
working and you're able to connect to the machine. It is a common
problem that networking isn't setup properly in these boxes.
Verify that authentication configurations are also setup properly,
as well.

If the box appears to be booting properly, you may want to increase
the timeout ("config.vm.boot_timeout") value.
  occurred at /home/purbon/logstash/qa/vagrant/command.rb:40:in `run!'

/home/purbon/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/net-ssh-2.9.4/lib/net/ssh/transport/session.rb:67:in `initialize': Object#timeout is deprecated, use Timeout.timeout instead.
/home/purbon/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/net-ssh-2.9.4/lib/net/ssh/transport/session.rb:84:in `initialize': Object#timeout is deprecated, use Timeout.timeout instead.
F/home/purbon/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/net-ssh-2.9.4/lib/net/ssh/transport/session.rb:67:in `initialize': Object#timeout is deprecated, use Timeout.timeout instead.
/home/purbon/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/net-ssh-2.9.4/lib/net/ssh/transport/session.rb:84:in `initialize': Object#timeout is deprecated, use Timeout.timeout instead.

An error occurred in an `after(:context)` hook.
  Net::SSH::Disconnect: connection closed by remote host
  occurred at /home/purbon/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/net-ssh-2.9.4/lib/net/ssh/transport/server_version.rb:50:in `rescue in block (2 levels) in negotiate!'
```

so the error is reproducible, but if this branch (this pr changes) is executed the flow is error less:

```bash
purbon@slave-342015:~/logstash/qa$ bundle exec rake qa:acceptance:single["ubuntu-1404"]
[Acceptance specs] running on ["ubuntu-1404"]

artifacts operation
  behaves like installable
    is installed on ubuntu-1404
    is running on ubuntu-1404
    is removable on ubuntu-1404
  behaves like updated
    can be updated an run on ubuntu-1404

CLI operation
  behaves like logstash version
    logstash --version
      on ubuntu-1404
        returns the right logstash version
  behaves like logstash install
    on ubuntu-1404
      with a direct internet connection
        when the plugin exist
          from a local `.GEM` file
....
```

so I'm +1 on merging this in and make the test stable until we're able to somehow overcome the snapshot/restore issue properly.

**Comment by purbon** [comment]
@ph thoughts on merging this due to my positive tests?

**Comment by ph** [comment]
LGTM, I've tested it this morning and the test didn't stop or timeout.

**Comment by ph** [comment]
+1 for the test on the ubuntu machine.

**Comment by purbon** [comment]
ok, will open an specific issue also to follow up on the snapshot issue, @dliappis might also help (he is a vagrant rock star) 👍 

squashing and shipping to have green test 👍

**Pull request closed without merging**